web/src/beta/components/fields/PropertyFields/index.tsx

```diff
@@ -18,31 +19,45 @@ const PropertyFields: React.FC<Props> = ({ propertyId, item }) => {
       {item?.schemaFields.map(sf => {
         const isList = item && "items" in item;
         const value = !isList ? item.fields.find(f => f.id === sf.id)?.value : sf.defaultValue;
-        return sf.type === "string" ? (
-          sf.ui === "color" ? (
-            <ColorField
-              key={sf.id}
-              name={sf.name}
-              value={(value as string) ?? ""}
-              description={sf.description}
-              onChange={handlePropertyValueUpdate(item.schemaGroup, propertyId, sf.id, sf.type)}
-            />
-          ) : sf.ui === "selection" || sf.choices ? (
-            <p key={sf.id}>Selection or choices field</p>
-          ) : sf.ui === "buttons" ? (
-            <p key={sf.id}>Button radio field</p>
-          ) : (
-            <TextInput
-              key={sf.id}
-              name={sf.name}
-              value={(value as string) ?? ""}
-              description={sf.description}
-              onChange={handlePropertyValueUpdate(item.schemaGroup, propertyId, sf.id, sf.type)}
-            />
-          )
-        ) : (
-          <p key={sf.id}>{sf.name} field</p>
-        );
+
+        switch (sf.type) {
```

Comment:
HmmMmmmMmmmMmmm thanks for giving this suggestion. I gave it some thought but still can't decide which is more readable....ternary operator is more condensed since we don't have to specify returns every time and also separate cases on individual lines...also, even with switch/case we would need to do some nesting, either way is not super easy to read.
Hmmm can you leave it as-is for now and we can come back to this later in a refactor or if it becomes hard to manage the more field components we add? 🙏🏻